### PR TITLE
fix: allow an empty requestBody to be provided for APIs that support …

### PR DESCRIPTION
**Pull request by bcoe**
This PR updates the generator such that an empty `requestBody` can be provided for methods that support `multipart` uploads (determined by checking `m.mediaUpload.protocols.simple.multipart`).

Most of the Google APIs I looked at require that the first section of the `multipart` POST represents metadata about the upload, e.g.,

> [Metadata part. Must come first, and must have a Content-Type header set to application/json; charset=UTF-8. Add the file's metadata to this part in JSON format](https://cloud.google.com/storage/docs/json_api/v1/how-tos/multipart-upload#sending_a_multipart_upload_request).

In testing with the youtube's `thumbnails.set`, the upstream API requires that empty metadata be set to perform a multi-part upload (it's my hunch that this behavior will be fairly consistent across APIs). You can now perform a multipart thumbnail upload like so:

```js
    await youtube.thumbnails.set(
      {
        videoId: 'abc123',
        requestBody: {},
        media: {
          mimeType: 'image/jpeg',
          body: fs.createReadStream(fileName),
        },
      },
      {
        onUploadProgress: (evt: {bytesRead: number}) => {
          progressEvents.push(evt.bytesRead);
        },
      }
    );
```

☝️ this will fire `onUploadProgress` events.

fixes: #1820

### CODE REVIEW

File: src/apis/accesscontextmanager/v1beta.ts [hunk 1]
```diff
@@ -449,19 +445,6 @@ export namespace accesscontextmanager_v1beta {
      */
     enableRestriction?: boolean | null;
   }
-  /**
-   * Alpha. Specifies how APIs are allowed to communicate within the Service Perimeter. This message is DEPRECATED and had been renamed to VpcAccessibleServices
-   */
-  export interface Schema$VpcServiceRestriction {
```
Comment (bcoe) [inline src/apis/accesscontextmanager/v1beta.ts]: most of the updates below are the result of running the generator, which tends to change a few existing APIs, regardless of changes to the templates.

File: src/apis/accesscontextmanager/v1beta.ts [hunk 2]
```diff
@@ -411,13 +411,9 @@ export namespace accesscontextmanager_v1beta {
      */
     unrestrictedServices?: string[] | null;
     /**
-     * Beta. Configuration for within Perimeter allowed APIs.
+     * Beta. Configuration for APIs allowed within Perimeter.
```
Comment (JustinBeckwith) [inline src/apis/accesscontextmanager/v1beta.ts]: Eugh.  Could I trouble you to first submit and land a re-generate PR from master?  Would love to see only the relevant changes in this PR.  Sorry.

File: test/test.media.ts [hunk 3]
```diff
@@ -142,6 +142,38 @@ describe('Media', () => {
...
+      )
+      .reply(200);
+    const fileName = path.join(__dirname, '../../test/fixtures/mediabody.txt');
+    const fileSize = fs.statSync(fileName).size;
+    const google = new GoogleApis();
+    const youtube = google.youtube('v3');
+    const progressEvents = new Array<number>();
+    await youtube.thumbnails.set(
+      {
+        videoId: 'abc123',
```
Comment (bcoe) [inline test/test.media.ts]: This test demonstrates how to perform a multipart upload with an API that doesn't expect a `requestBody`.

File: src/generator/templates/api-endpoint.njk [hunk 4]
```diff
@@ -154,6 +154,11 @@ export interface Schema${{ schema.id }} {
           * Request body metadata
           */
           requestBody?: Schema${{ m.request.$ref }};
+      {% elif m.supportsMediaUpload and m.mediaUpload.protocols.simple.multipart %}
```
Comment (bcoe) [inline src/generator/templates/api-endpoint.njk]: this is the meat of the change.

**Pull request merged**